packages/next/src/shared/lib/get-img-props.ts

```diff
@@ -465,6 +465,16 @@ export function getImgProps(
             `Image with src "${src}" has invalid "height" property. Expected a numeric value in pixels but received "${height}".`
           )
         }
+        if (src.startsWith(' ')) {
+          throw new Error(
+            `Image with src "${src}" cannot start with a space. Consider using src.trimStart() to remove it.`
+          )
+        }
+        if (src.endsWith(' ')) {
+          throw new Error(
+            `Image with src "${src}" cannot end with a space. Consider using src.trimEnd() to remove it or replace it with "%20" to keep it.`
```

Comment:
From the PR description it sounds like it can be used with surrounding spaces. Just that they get automatically removed. Should we maybe warn instead? Seems harsh to hard error here.

Reply:
Warnings are often missed and that would mean a developer could deploy their app and not know until runtime (when a request is made to the optimizer) that the request will fail. Now their app is broken in production. 

It’s best to fail fast while in development to catch this corner case much earlier.

Reply:
This sounds like you're arguing against any warnings?

Reply:
Correct, I am against warnings and for errors because it will fail fast.